src/pages/ocpDetails/detailsTable.tsx

```diff
@@ -0,0 +1,327 @@
+import {
+  sortable,
+  SortByDirection,
+  Table,
+  TableBody,
+  TableHeader,
+} from '@patternfly/react-table';
+import { getQuery, OcpQuery } from 'api/ocpQuery';
+import { OcpReport } from 'api/ocpReports';
+import React from 'react';
+import { InjectedTranslateProps, translate } from 'react-i18next';
+import { connect } from 'react-redux';
+import { formatCurrency } from 'utils/formatValue';
+import {
+  getIdKeyForGroupBy,
+  getUnsortedComputedOcpReportItems,
+} from 'utils/getComputedOcpReportItems';
+import { ComputedOcpReportItem } from 'utils/getComputedOcpReportItems';
+import {
+  monthOverMonthOverride,
+  styles,
+  tableOverride,
+} from './detailsTable.styles';
+import { DetailsTableItem } from './detailsTableItem';
+
+interface DetailsTableOwnProps {
+  onSelected(selectedItems: ComputedOcpReportItem[]);
+  onSort(value: string, isSortAscending: boolean);
+  query: OcpQuery;
+  report: OcpReport;
+}
+
+interface State {
+  columns?: any[];
+  isHistoricalModalOpen?: boolean;
+  rows?: any[];
+}
+
+type DetailsTableProps = DetailsTableOwnProps & InjectedTranslateProps;
+
+class DetailsTableBase extends React.Component<DetailsTableProps> {
+  public state: State = {
+    columns: [],
+    isHistoricalModalOpen: false,
+    rows: [],
+  };
+
+  constructor(props: DetailsTableProps) {
+    super(props);
+    this.handleOnCollapse = this.handleOnCollapse.bind(this);
+    this.handleOnSelect = this.handleOnSelect.bind(this);
+    this.handleOnSort = this.handleOnSort.bind(this);
+  }
+
+  public componentDidMount() {
+    this.setState({});
+  }
+
+  public componentDidUpdate(prevProps: DetailsTableProps) {
+    const { query, report } = this.props;
+    if (
+      report !== prevProps.report ||
+      getQuery(query) !== getQuery(prevProps.query)
+    ) {
+      this.initDatum();
+    }
+  }
+
+  private initDatum = () => {
+    const { query, report, t } = this.props;
+    if (!query || !report) {
+      return;
+    }
+
+    const groupById = getIdKeyForGroupBy(query.group_by);
+    const total =
+      report && report.total ? formatCurrency(report.total.charge) : 0;
+
+    const columns = [
+      {
+        orderBy: groupById,
+        title: t('ocp_details.name_column_title', { groupBy: groupById }),
+        transforms: [sortable],
+      },
+      {
+        title: t('ocp_details.change_column_title'),
+      },
+      {
+        orderBy: 'charge',
+        title: t('ocp_details.charge_column_title', { total }),
+        transforms: [sortable],
+      },
+    ];
+
+    const rows = [];
+    const computedItems = getUnsortedComputedOcpReportItems({
+      report,
+      idKey: groupById,
+    });
+
+    computedItems.map((item, index) => {
+      rows.push(
+        {
+          cells: [
+            item.label,
+            this.getMonthOverMonthCost(item, index),
+            this.getTotalCost(item, index),
+          ],
+          isOpen: false,
+          item,
+        },
+        {
+          parent: index,
```

Comment:
This is wrongly calculated parent. What you end up with is
```JS
[
 { cells: ['parent - 1'], isOpen: false }, // 0
 { cells: ['child - 1'], parent: 0}, // 1
 { cells: ['parent - 2'], isOpen: false }, // 2
 { cells: ['child - 2'], parent: 1 } // this is wrong, it should be number 2
]
```

To calculate parent correctly you need to use `index * 2`

Reply:
Ah, much better. Thank you!